src/services/kuzzleWrapper-v2.ts

```diff
@@ -0,0 +1,284 @@
+    return getValueAdditionalAttribute(content[attribute], attributePath)
+  }
+
+  return content[attribute]
+}
+
+const formatMeta = _kuzzle_info => ({
+  author: _kuzzle_info.author === '-1' ? 'Anonymous (-1)' : _kuzzle_info.author,
+  updater:
+    _kuzzle_info.updater === '-1' ? 'Anonymous (-1)' : _kuzzle_info.updater,
+  createdAt: _kuzzle_info.createdAt,
+  updatedAt: _kuzzle_info.updatedAt
+})
+
+export const performSearchDocuments = async (
+  collection,
+  index,
+  filters = { query: {} },
+  pagination = {},
+  sort = []
+) => {
+  if (!collection || !index) {
+    throw new Error('Missing collection or index')
+  }
+
+  const result = await kuzzle.document.search(
+    index,
+    collection,
+    { ...filters, sort },
+    { ...pagination }
+  )
+
+  const documents = result.hits.map(d => ({
+    id: d._id,
+    ...d._source,
+    _kuzzle_info: d._source._kuzzle_info
+      ? formatMeta(d._source._kuzzle_info)
+      : undefined
+  }))
+
+  const totalDocument = await kuzzle.document.count(index, collection, {
+    query: filters.query
+  })
+
+  return { documents, total: totalDocument }
+}
+
+export const getMappingDocument = async (collection, index) => {
+  // @todo Use the SDK method after
+  // https://github.com/kuzzleio/sdk-javascript/pull/507 is merged
+  const request = {
+    controller: 'collection',
+    action: 'getMapping',
+    index,
+    collection,
+    includeKuzzleMeta: true
+  }
+
+  const response = await kuzzle.query(request)
+
+  return response.result
+}
+
+export const performDeleteDocuments = async (index, collection, ids) => {
+  if (
+    !ids ||
+    !Array.isArray(ids) ||
+    ids.length === 0 ||
+    !index ||
+    !collection
+  ) {
+    throw new Error('ids<Array> parameter is required')
+  }
+
+  const deleted = await kuzzle.document.mDelete(index, collection, ids, {
+    refresh: 'wait_for'
+  })
+
+  return deleted
+}
+
+// ### Security
+
+// Users related
+export const performSearchUsers = async (
+  collection,
+  index,
+  filters = {},
+  pagination = {},
+  sort = []
+) => {
+  const strategies = await kuzzle.auth.getStrategies()
+
+  const result = await kuzzle.security.searchUsers(
+    { ...filters, sort },
+    { ...pagination }
+  )
+  let additionalAttributeName: any = null
+
+  if (sort.length > 0) {
+    if (typeof sort[0] === 'string') {
+      additionalAttributeName = sort[0]
+    } else {
+      additionalAttributeName = Object.keys(sort[0])[0]
+    }
+  }
+
+  const users: Array<any> = []
+  for (const d of result.hits) {
+    const u: any = {
+      id: d._id,
+      ...d.content,
+      _kuzzle_info: formatMeta(d.content._kuzzle_info),
+      credentials: {}
+    }
+    for (const strategy of strategies) {
+      try {
+        const res = await kuzzle.security.getCredentials(strategy, d._id)
+        u.credentials[strategy] = res
+      } catch (e) {
+        u.credentials[strategy] = {}
+      }
+    }
+
+    users.push(u)
+  }
```

Comment:
Is the[ id-length](https://eslint.org/docs/rules/id-length) of this project at 1? (d, u) Kinda cryptic.

Reply:
ROTFL! Ur right dude